File: frontend/src/app/features/dashboard/keys/dtos/key-response.dto.ts
import {BaseResponseDto} from '../../../../core/dtos/base-response.dto';
import {RoomResponseDto} from '../../rooms/dtos/room-response.dto';
import {KeyStatus} from '../enums/key.enum';

export interface KeyResponseDto extends BaseResponseDto {
  code: string;
  room: RoomResponseDto,
  status: KeyStatus;
}

File: backend/src/main/java/io/hb3nce04/keykeeperapp/feature/key/model/dto/response/KeyResponseDto.java
package io.hb3nce04.keykeeperapp.feature.key.model.dto.response;

import io.hb3nce04.keykeeperapp.feature.key.model.enums.KeyStatus;
import io.hb3nce04.keykeeperapp.feature.room.model.dto.RoomResponseDto;
import io.hb3nce04.keykeeperapp.common.model.dto.BaseDto;
import lombok.Getter;
import lombok.Setter;

@Getter
@Setter
public class KeyResponseDto extends BaseDto {
    private String code;

    private RoomResponseDto room;

    private KeyStatus status;
}
